ycmd/completers/cs/cs_completer.py

```diff
@@ -545,30 +699,279 @@ def ServerTerminated( self ):
              self._omnisharp_phandle.poll() is not None )
 
 
-  def _SolutionFile( self ):
-    """ Find out which solution file server was started with """
-    return self._solution_path
+  def DebugInfo( self, request_data ):
+    """ Get debug info."""
+    if self._separate_log_file:
+      return 'OmniSharp logfiles:\n{0}\n{1}'.format(
+                    self._filename_stdout,
+                    self._filename_stderr )
+    else:
+      return "Omnisharp logs: included in ycmd logs"
+
+
+  def _ForceStopServer( self ):
+    # Kill it if it's still up
+    phandle = self._omnisharp_phandle
+    if phandle is not None:
+      self._logger.info( 'Killing OmniSharp server' )
+      for stream in [ phandle.stderr, phandle.stdout ]:
+        if stream is not None:
+          stream.close()
+      try:
+        phandle.kill()
+      except OSError as e:
+        if e.errno == 3:
```

Comment:
3 is the best number. :)

In other words, you need to use a named constant here. The number alone is way too magic.